src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -0,0 +1,322 @@
+package org.folio.circulation.resources;
+
+import static org.folio.circulation.domain.representations.RequestProperties.PROXY_USER_ID;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.ValidationErrorFailure.failedValidation;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import org.folio.circulation.domain.CreateRequestService;
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.ItemStatus;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.domain.LocationRepository;
+import org.folio.circulation.domain.RequestAndRelatedRecords;
+import org.folio.circulation.domain.RequestQueueRepository;
+import org.folio.circulation.domain.RequestRepository;
+import org.folio.circulation.domain.RequestRepresentation;
+import org.folio.circulation.domain.RequestType;
+import org.folio.circulation.domain.ServicePointRepository;
+import org.folio.circulation.domain.UpdateItem;
+import org.folio.circulation.domain.UpdateLoan;
+import org.folio.circulation.domain.UpdateLoanActionHistory;
+import org.folio.circulation.domain.UserRepository;
+import org.folio.circulation.domain.policy.LoanPolicyRepository;
+import org.folio.circulation.domain.policy.RequestPolicyRepository;
+import org.folio.circulation.domain.representations.RequestByInstanceIdRequest;
+import org.folio.circulation.domain.validation.ProxyRelationshipValidator;
+import org.folio.circulation.domain.validation.ServicePointPickupLocationValidator;
+import org.folio.circulation.storage.ItemByInstanceIdFinder;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CreatedJsonResponseResult;
+import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.Pair;
+import org.folio.circulation.support.ResponseWritableResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.ServerErrorFailure;
+import org.folio.circulation.support.http.server.WebContext;
+import org.joda.time.format.ISODateTimeFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class RequestByInstanceIdResource extends Resource {
+  private UserRepository userRepository;
```

Comment:
Why are these declared as fields rather than variables within the method?

I don't know vert.x well, so I might be wrong about this. This class is only instantiated once. If there are interleaving HTTP requests being handled I imagine they could interfere with each other by changing this shared state.

Reply:
These are fields because they are being used elsewhere and I didn't want to pass them around. I don't know vertx much so if it's better to declare them within the method then I will do that.

Reply:
My belief would be that state could be corrupted if multiple requests come in at the same time. Does my example make sense, that a second HTTP request will change the variables and hence affect the first?

I think it would be safer to use variables within the method. I'm aware that leads to noisier code.